### DISCUSSION TITLE
How do I configure CSP?

### DISCUSSION THREAD

**Discussion by jiazengp** [vuepress/core#241] [Q&A] [answered]
such as
``` html
<meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src https://*; child-src 'none';">
```

**Comment by yzx9** [comment, marked as answer]
It’s easy to set it up yourself in user config, follow the config in the official document:

https://github.com/vuepress/vuepress-next/blob/ca22d4fe7ade66571f34dc80343f4ec57483b44a/docs/.vuepress/config.ts#L8-L51

We can config like this:

```ts
export default defineUserConfig<DefaultThemeOptions>({ 
   head: [ 
     [ 
       'meta', 
       { 
         "http-equiv": 'Content-Security-Policy', 
         content: "default-src 'self'; img-src https://*; child-src 'none';", 
       }, 
     ], 
  ]
  ...otherConfig
}
```

**Reply by gspgsp** [reply]
i don't know if this will work, i meet the same problem.

**Reply by yzx9** [reply]
Aha, even though a long time has passed, I guess it still works. Refer to [VuePress Config Documentation](https://vuepress.vuejs.org/reference/config.html#head).

**Reply by gspgsp** [reply]
oh, i add these config, but it will not work,  the problem for me is that:  i define a plugin, and it contains many ts file, and global variables, when i comment out this plugin, the site work well.

**Reply by Mister-Hope** [reply]
Then that's the plugin's problem
